images/airflow/2.9.2/python/mwaa/celery/task_monitor.py

```diff
@@ -55,21 +57,34 @@
 )
 DEFAULT_QUEUE_ENV_KEY = "AIRFLOW__CELERY__DEFAULT_QUEUE"
 
-# Allow at least 3 minutes for the worker to warm up before checking for idleness or
-# cleaning abandoned resources.
+# A worker maybe be busy loading the required libraries or polling messages from the environment SQS queue, so we allow the worker to
+# warm up before checking for idleness or cleaning abandoned resources.
 IDLENESS_CHECK_WARMUP_WAIT_PERIOD = timedelta(minutes=3)
 CLEANUP_ABANDONED_RESOURCES_WARMUP_WAIT_PERIOD = timedelta(minutes=3)
-# Allow at least 1 second between any two consecutive idleness checks.
+# If the idleness check is made too aggressively, then we will be reusing the result of the previous check till
+# the check delay threshold is reached.
 IDLENESS_CHECK_DELAY_PERIOD = timedelta(seconds=1)
-# The worker should be idle for at least 2 consecutive idleness check before being
-# declared idle.
+# The worker should be idle for some consecutive checks before being declared idle.
 CONSECUTIVE_IDLENESS_CHECK_THRESHOLD = 2
-# Allow at least 1 minute between any two consecutive abandoned resource cleanups.
+# The monitor is also responsible for performing cleanups/corrections of in-memory state in case of issues such as inability to
+# terminate the Airflow task process once it has finished or deleting the message from the environment SQS queue. This process requires
+# scanning of multiple shared memory blocks and possibly executing SQS operations, so we do this only after a time threshold since
+# the last cleanup is breached.
 CLEANUP_ABANDONED_RESOURCES_DELAY_PERIOD = timedelta(minutes=1)
-# If in the improbable case of the worker picking up new tasks after having paused its
-# consumption, we reset the worker to non-idle state and backoff from checking for
-# idleness for a minute.
+# If in the improbable case of the worker picking up new tasks after having paused its consumption, we reset the worker to non-idle
+# state and backoff from checking for idleness for a certain threshold.
 IDLENESS_RESET_BACKOFF_PERIOD = timedelta(minutes=1)
+# In case of issues, signals can arrive late and out of order. So, we scan all unprocessed signals within a time range.
+# Only a handful of signals are expected for each worker, so this repeated processing should be very light.
+SIGNAL_SEARCH_TIME_RANGE = timedelta(hours=1)
```

Comment:
I think I understand. Here is my rationale behind the current state of things: the main and sidecar container each have their own set of responsibility and each is given 1 hour to ensure that they are able to finish their part. The 1 hour limit starts from the point in time of creation of their corresponding file and not the creation of the signal, so issues in sidecar container does not reduce the time given to the main container to finish its processing. This means that this config does not need to be in line with the sidecar config and it can be changed independently if needed.

Reply:
Ah, I do think that the sidecar and main container can operate independent of this config, so I think this is fine. I was confusing myself between the EFS and EBS volumes.